unit_testing/cli/test_cli.py

```diff
@@ -0,0 +1,27 @@
+from spinalcordtoolbox.utils import __sct_dir__
+import os
+import pytest
+
+scripts = []
+for line in open(os.path.join(__sct_dir__, 'scripts.txt')):
+    li = line.strip()
+    if not li.startswith("#"):
+        scripts.append(li)
+
+# Exclude scripts where no arguments is valid usage
+no_args_scripts = [s for s in scripts if s not in [
+    'isct_test_ants',          # No args -> tests ants binaries
+    'sct_testing',             # No args -> runs full test suite
+    'sct_check_dependencies',  # No args -> checks dependencies
+    'sct_version'              # No args -> prints version
+]]
```

Comment:
Could these be `@fixture`s?

Reply:
Loading test cases from a file dynamically is the case that pytest *doesn't* support natively. So... I guess not really.

Reply:
Well, it's the exact same problem we're facing in #3152, eh?

Test collection dependent on loading files. :P

Reply:
I don't really think it's the same. This one would have collection dependent on loading data, that one just has the test execution dependent on loading data. (which is why I am pushing to use fixtures in that one: we *should* be able to without causing pytest any problems)